sample-apps/donet-sample-app/Controllers/MetricEmitter.cs

```diff
@@ -0,0 +1,137 @@
+using System;
+using System.Collections.Generic;
+using OpenTelemetry;
+using System.Diagnostics.Metrics;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Instrumentation;
+
+namespace integration_test_app.Controllers
+{
+    public class MetricEmitter
+    {
+        const string DIMENSION_API_NAME = "apiName";
+        const string DIMENSION_STATUS_CODE = "statusCode";
+        
+
+        static string API_COUNTER_METRIC = "totalApiRequests";
+        static string API_LATENCY_METRIC = "latencyTime";
+        static string API_SUM_METRIC = "totalBytesSent";
+        static string API_TOTAL_TIME_METRIC = "timeAlive";
+        static string API_TOTAL_HEAP_SIZE = "heapSize";
+        static string API_TOTAL_THREAD_SIZE = "threadSize";
+        static string API_CPU_USAGE = "cpuUsage";
+    
+
+        Histogram<double> apiLatencyRecorder;
+        Counter<long> totalTimeSentObserver;
+
+        static long apiRequestSent = 0;
+        static long totalBytesSent = 0;
+        static long totalHeapSize  = 0;
+        static long totalThreadSize = 0;
+        static int cpuUsage = 0;
+
+        // The below API name and status code dimensions are currently shared by all metrics observer in
+        // this class.
+        string apiNameValue = "";
+        string statusCodeValue = "";
+
+        
+        public MetricEmitter()
+        {
+            Meter meter = new Meter("adot", "1.0");
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter("adot")
+                .AddOtlpExporter()
+                .Build();
+            
+            string latencyMetricName = API_LATENCY_METRIC;
+            string totalApiRequestSent = API_COUNTER_METRIC;
+            string totalApiBytesSentMetricName = API_SUM_METRIC;
+            string totaltimealiveMetricName = API_TOTAL_TIME_METRIC;
+            string totalHeapSizeMetricName = API_TOTAL_HEAP_SIZE;
+            string totalThreadsMetricName = API_TOTAL_THREAD_SIZE;
+            string cpuUsageMetricName = API_CPU_USAGE;
+
+
+            string instanceId = Environment.GetEnvironmentVariable("INSTANCE_ID");
```

Comment:
Why do we need this? The application name is a resource attribute and it will be added as attribute to all metrics. You can still define extra resource attributes like this id. The name of the metric should be predictable and we should add extra data through attributes

Reply:
The new spec changes has mandated to append the instanceId (if it exists) to metric name.